processor/src/lib.rs

```diff
@@ -137,7 +137,8 @@ where
 // PROCESS
 // ================================================================================================
 
-pub struct Process<A>
+#[cfg(not(feature = "internals"))]
```

Comment:
question: why not using the pattern with the private module and option pub use?

I'm asking because I assume that version would have less code duplication, namely the `Process` struct would be defined only once. And line of codes was one of the motivations for that suggestion.

Reply:
I believe your proposal was to do something like the following.  I think this would work and is definitely a pattern that we can consider but I don't think it's as readable personally.  Happy to go with the consensus on this.

```rust
#[cfg(feature = "internals")]
pub use processor_internal::Process;

#[cfg(not(feature = "internals"))]
use processor_internal::Process;

mod processor_internal {
    use super::*;

    pub struct Process<A>
    where
        A: AdviceProvider,
    {
        pub system: System,
        pub decoder: Decoder,
        pub stack: Stack,
        pub range: RangeChecker,
        pub chiplets: Chiplets,
        pub advice_provider: A,
    }
}
```